blog/test.py

```diff
@@ -11,75 +12,52 @@
 from trips import views
 
 
-class ArticleTest(TestCase):
-    def setUp(self):
-        self.client = app.test_client()
-        self.user = User(email='emailtest@test.com').set_password(
-            'testtest').save()
-
-    def tearDown(self):
-        User.objects.delete()
-        Article.objects.delete()
-
+class ArticleTest(BaseTest):
     def test_articles_empty(self):
         result = self.client.get('/articles')
         self.assertEqual(result.status_code, 200)
 
     def test_articles_with_an_article(self):
-        article = Article(title='<h1>title</h1>', content='<p>content</p>')
-        article.author = self.user
-        article.save()
+        Article(title='title',
+                content='<p>content</p>',
+                author=self.create_user()).save()
         result = self.client.get('/articles')
         self.assertIn(b'<article class=preview>', result.data)
 
     def test_read_article_unauthenticated(self):
-        article = Article(title='<h1>title</h1>', content='<p>content</p>')
-        article.author = self.user
-        article.save()
+        article = Article(title='title',
+                          content='<p>content</p>',
+                          author=self.create_user()).save()
         result = self.client.get('/@emailtest/{}-{}'.format(article.slug,
                                                             str(article.id)))
-        self.assertEqual(result.status_code, 301)
+        self.assertEqual(result.status_code, 200)
 
     def test_read_article_authenticated(self):
-        data = {
-            'email': 'emailtest@test.com',
-            'password': 'testtest',
-        }
-        self.client.post('/login', data=data, follow_redirects=True)
-        article = Article(title='title', content='<p>content</p>')
-        article.author = self.user
-        article.save()
+        user = self.create_user()
```

Comment:
That is because you do a `self.user` and then you try to recreate the same user with the same data in the same test. Once `self.user` exists, you should check that you're not duplicating this user ;)